homeassistant/components/growatt_server/coordinator.py

```diff
@@ -0,0 +1,206 @@
+"""Coordinator module for managing Growatt data fetching."""
+
+import datetime
+import json
+import logging
+from typing import Any
+
+import growattServer
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util import dt as dt_util
+
+from .const import DEFAULT_URL, DOMAIN
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GrowattCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Coordinator to manage Growatt data fetching."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        device_id: str,
+        device_type: str,
+        plant_id: str,
+    ) -> None:
+        """Initialize the coordinator."""
+        self.config = {**config_entry.data}
```

Comment:
I am not sure why we need to store this.

Let's overwrite the config entry type on the coordinator, then we can just freely access `coordinator.config_entry.data`